src/modules/rover_differential/RoverDifferential.cpp

```diff
@@ -0,0 +1,256 @@
+				_differential_setpoint.throttle = manual_control_setpoint.throttle;
+				_differential_setpoint.yaw_rate = math::interpolate<float>(manual_control_setpoint.roll,
+								  -1.f, 1.f,
+								  -_max_yaw_rate, _max_yaw_rate);
+			}
+
+			_differential_setpoint.closed_loop_yaw_rate = true;
+		} break;
+
+	case vehicle_status_s::NAVIGATION_STATE_AUTO_MISSION:
+	case vehicle_status_s::NAVIGATION_STATE_AUTO_RTL:
+		_differential_setpoint = _rover_differential_guidance.computeGuidance(_vehicle_yaw, _vehicle_forward_speed,
+					 _nav_state);
+		break;
+
+	default: // Unimplemented nav states will stop the rover
+		_differential_setpoint.throttle = 0.f;
+		_differential_setpoint.yaw_rate = 0.f;
+		_differential_setpoint.closed_loop_yaw_rate = false;
+		break;
+	}
+
+	float speed_diff_normalized = _differential_setpoint.yaw_rate;
+
+	// Closed loop yaw rate control
+	if (_differential_setpoint.closed_loop_yaw_rate) {
+		if (fabsf(_differential_setpoint.yaw_rate - _vehicle_body_yaw_rate) < YAW_RATE_ERROR_THRESHOLD) {
+			speed_diff_normalized = 0.f;
+			pid_reset_integral(&_pid_yaw_rate);
+
+		} else {
+			const float speed_diff = _differential_setpoint.yaw_rate * _param_rd_wheel_track.get(); // Feedforward
+			speed_diff_normalized = math::interpolate<float>(speed_diff, -_param_rd_max_speed.get(),
+						_param_rd_max_speed.get(), -1.f, 1.f);
+			speed_diff_normalized = math::constrain(speed_diff_normalized +
+								pid_calculate(&_pid_yaw_rate, _differential_setpoint.yaw_rate, _vehicle_body_yaw_rate, 0, dt),
+								-1.f, 1.f); // Feedback
+		}
+
+	} else {
+		pid_reset_integral(&_pid_yaw_rate);
+	}
+
+	// Publish rover differential status (logging)
+	rover_differential_status_s rover_differential_status{};
+	rover_differential_status.timestamp = _timestamp;
+	rover_differential_status.actual_speed = _vehicle_forward_speed;
+	rover_differential_status.desired_yaw_rate_deg_s = M_RAD_TO_DEG_F * _differential_setpoint.yaw_rate;
+	rover_differential_status.actual_yaw_rate_deg_s = M_RAD_TO_DEG_F * _vehicle_body_yaw_rate;
+	rover_differential_status.pid_yaw_rate_integral = _pid_yaw_rate.integral;
+	_rover_differential_status_pub.publish(rover_differential_status);
+
+	// Publish to motors
+	actuator_motors_s actuator_motors{};
+	actuator_motors.reversible_flags = _param_r_rev.get();
+	computeMotorCommands(_differential_setpoint.throttle, speed_diff_normalized).copyTo(actuator_motors.control);
+	actuator_motors.timestamp = _timestamp;
+	_actuator_motors_pub.publish(actuator_motors);
+
+}
+
+matrix::Vector2f RoverDifferential::computeMotorCommands(float forward_speed, const float speed_diff)
+{
+	float combined_velocity = fabsf(forward_speed) + fabsf(speed_diff);
+
+	if (combined_velocity > 1.0f) { // Prioritize yaw rate
+		float excess_velocity = fabsf(combined_velocity - 1.0f);
+		forward_speed -= sign(forward_speed) * excess_velocity;
+	}
+
+	// Calculate the left and right wheel speeds
+	return Vector2f(forward_speed - speed_diff,
```

Comment:
Shouldn't a positive yaw rate setpoint (speed_diff) result in an increase on the left and a reduction on the right?

Reply:
(I assume ActuatorMotors(_left_, _right_)

Reply:
It seems that the definition is the other way around ActutorMotors(right, left)
![image](https://github.com/user-attachments/assets/d9f38b73-2d33-4e20-8270-630ba7599cfe)
It also works as intended on hardware tests.